design/src/main/java/com/ingresse/design/ui/user/DSUserInfoCard.kt

```diff
@@ -0,0 +1,70 @@
+package com.ingresse.design.ui.user
+
+import android.content.Context
+import android.util.AttributeSet
+import android.view.View
+import androidx.constraintlayout.widget.ConstraintLayout
+import com.ingresse.design.R
+import com.ingresse.design.helper.ResourcesHelper
+import com.ingresse.design.helper.USER_IMAGE_PREFIX
+import com.ingresse.design.helper.setVisible
+import kotlinx.android.synthetic.main.ds_user_info_card.view.*
+
+class DSUserInfoCard(context: Context, attrs: AttributeSet): ConstraintLayout(context, attrs) {
+    val resHelper = ResourcesHelper(context)
+    var type: DSUserInfoCardType = DSUserInfoCardType.UNDEFINED
+    set(value) {
+        field = value
+        updateButton()
+    }
+
+    init {
+        inflate(context, R.layout.ds_user_info_card, this)
+        val array = context.theme.obtainStyledAttributes(attrs,
+            R.styleable.DSUserInfosCard, 0, 0)
+
+        val typeId = array.getInt(R.styleable.DSUserInfosCard_userCardType,
+            DSUserInfoCardType.UNDEFINED.id)
+
+        type = DSUserInfoCardType.getById(typeId)
+        updateButton()
+
+        array.recycle()
+    }
+
+    /**
+     * Function to set the button action
+     *
+     * @param onClick - Action who will called when click on button
+     */
+    fun onButtonClick(onClick: (View) -> Unit) { btn_action.setOnClickListener(onClick) }
+
+    /**
+     * Function to set the user information
+     *
+     * @param name - User's name. If is empty label not will show.
+     * @param email - User's email. If is empty label not will show.
+     * @param address - User's address. If is empty label not will show.
```

Comment:
Aqui fica só uma observação para gramática em inglês. O uso de negação tá incorreto. Você pode também trocar as palavras para evitar o uso de gramática mais complexa, tal como _label is hidden_ ou algo do gênero.